### PR TITLE
Add RepString from Base

### PR DESCRIPTION
**Pull request by ararslan**
Ref https://github.com/JuliaLang/julia/issues/19564

### CODE REVIEW

File: src/LegacyStrings.jl [hunk 1]
```diff
@@ -81,4 +82,10 @@ import Base:
     else
         using Base: UTF_ERR_SHORT, checkstring
     end
+
+    if VERSION >= v"0.6.0"
```
Comment (tkelman) [inline src/LegacyStrings.jl]: should use isdefined

Comment (ararslan) [inline src/LegacyStrings.jl]: Good call, thanks!

**Comment by ararslan** [comment]
The failure on 32-bit Windows nightly is a timeout.

**Comment by StefanKarpinski** [comment]
LGTM – I assume you just copied the code from Base + the relevant tests?

**Comment by ararslan** [comment]
>  assume you just copied the code from Base + the relevant tests?

Yep. I didn't copy the definition of `repeat` that returns a `RepString` though. Should I?

**Comment by stevengj** [comment]
LGTM.

**Comment by stevengj** [comment]
Are there docstrings that need to be copied over?

**Comment by stevengj** [comment]
Also, the PR should update README.md

**Comment by ararslan** [comment]
> Are there docstrings that need to be copied over?

Nope.

> Also, the PR should update README.md

Good call, will do. Thanks!

**Comment by ararslan** [comment]
I just realized that I forgot to make the corresponding Base PR that removes `RepString`. Now that this has been merged and the package has been tagged to include it, I'll do that.

**Comment by tkelman** [comment]
Jeff beat you to it https://github.com/JuliaLang/julia/pull/19867 (which is why I went and tagged this)

**Pull request merged**